### PR TITLE
Add more resources

### PR DESCRIPTION
**Pull request by aliscott**
**Objective**:

Add support for S3 and Route 53.

**Example output**:

Route 53:
```
  aws_route53_record.alias2
  └─ Standard queries                                   -  1M queries   0.4000            -             -
  Total                                                                                   -             -

  aws_route53_record.geo
  └─ Geo DNS queries                                    -  1M queries   0.7000            -             -
  Total                                                                                   -             -

  aws_route53_record.latency
  └─ Latency based routing queries                      -  1M queries   0.6000            -             -
  Total                                                                                   -             -

  aws_route53_record.standard
  └─ Standard queries                                   -  1M queries   0.4000            -             -
  Total                                                                                   -             -

  aws_route53_zone.zone1
  └─ Per hosted zone                                    1  months       0.5000       0.0007        0.5000
  Total                                                                              0.0007        0.5000
```

S3:
```
  aws_s3_bucket.bucket1
  ├─ Object tagging                                                -  10k tags      0.0100            -             -
  ├─ Standard
  │  ├─ Storage                                                    -  GB            0.0230            -             -
  │  ├─ PUT, COPY, POST, LIST requests                             -  1k requests   0.0050            -             -
  │  ├─ GET, SELECT, and all other requests                        -  1k requests   0.0004            -             -
  │  ├─ Select data scanned                                        -  GB            0.0020            -             -
  │  └─ Select data returned                                       -  GB            0.0007            -             -
  ├─ Glacier deep archive
  │  ├─ Storage                                                    -  GB            0.0010            -             -
  │  ├─ PUT, COPY, POST, LIST requests                             -  1k requests   0.0500            -             -
  │  ├─ GET, SELECT, and all other requests                        -  1k requests   0.0004            -             -
  │  ├─ Lifecycle transition                                       -  1k requests   0.0500            -             -
  │  ├─ Retrieval requests (standard)                              -  1k requests   0.1000            -             -
  │  ├─ Retrievals (standard)                                      -  GB            0.0200            -             -
  │  ├─ Retrieval requests (bulk)                                  -  1k requests   0.0250            -             -
  │  ├─ Retrievals (bulk)                                          -  GB            0.0025            -             -
  │  └─ Early delete (within 180 days)                             -  GB            0.0010            -             -
  ├─ Glacier
  │  ├─ Storage                                                    -  GB            0.0040            -             -
  │  ├─ PUT, COPY, POST, LIST requests                             -  1k requests   0.0500            -             -
  │  ├─ GET, SELECT, and all other requests                        -  1k requests   0.0004            -             -
  │  ├─ Lifecycle transition                                       -  1k requests   0.0500            -             -
  │  ├─ Retrieval requests (standard)                              -  1k requests   0.0500            -             -
  │  ├─ Retrievals (standard)                                      -  GB            0.0100            -             -
  │  ├─ Select data scanned (standard)                             -  GB            0.0080            -             -
  │  ├─ Select data returned (standard)                            -  GB            0.0100            -             -
  │  ├─ Retrieval requests (expedited)                             -  1k requests  10.0000            -             -
  │  ├─ Retrievals (expedited)                                     -  GB            0.0300            -             -
  │  ├─ Select data scanned (expedited)                            -  GB            0.0200            -             -
  │  ├─ Select data returned (expedited)                           -  GB            0.0300            -             -
  │  ├─ Retrieval requests (bulk)                                  -  1k requests   0.0250            -             -
  │  ├─ Retrievals (bulk)                                          -  GB            0.0025            -             -
  │  ├─ Select data scanned (bulk)                                 -  GB            0.0010            -             -
  │  ├─ Select data returned (bulk)                                -  GB            0.0025            -             -
  │  └─ Early delete (within 90 days)                              -  GB            0.0040            -             -
  ├─ Intelligent tiering
  │  ├─ Storage (frequent access)                                  -  GB            0.0230            -             -
  │  ├─ Storage (infrequent access)                                -  GB            0.0125            -             -
  │  ├─ Monitoring and automation                                  -  1k objects    0.0025            -             -
  │  ├─ PUT, COPY, POST, LIST requests                             -  1k requests   0.0050            -             -
  │  ├─ GET, SELECT, and all other requests                        -  1k requests   0.0004            -             -
  │  ├─ Lifecycle transition                                       -  1k requests   0.0100            -             -
  │  ├─ Select data scanned                                        -  GB            0.0020            -             -
  │  ├─ Select data returned                                       -  GB            0.0007            -             -
  │  └─ Early delete (within 30 days)                              -  GB            0.0230            -             -
  ├─ One zone - infrequent access
  │  ├─ Storage                                                    -  GB            0.0100            -             -
  │  ├─ PUT, COPY, POST, LIST requests                             -  1k requests   0.0100            -             -
  │  ├─ GET, SELECT, and all other requests                        -  1k requests   0.0010            -             -
  │  ├─ Lifecycle transition                                       -  1k requests   0.0100            -             -
  │  ├─ Retrievals                                                 -  GB            0.0100            -             -
  │  ├─ Select data scanned                                        -  GB            0.0020            -             -
  │  └─ Select data returned                                       -  GB            0.0100            -             -
  └─ Standard - infrequent access
     ├─ Storage                                                    -  GB            0.0125            -             -
     ├─ PUT, COPY, POST, LIST requests                             -  1k requests   0.0100            -             -
     ├─ GET, SELECT, and all other requests                        -  1k requests   0.0010            -             -
     ├─ Lifecycle transition                                       -  1k requests   0.0100            -             -
     ├─ Retrievals                                                 -  GB            0.0100            -             -
     ├─ Select data scanned                                        -  GB            0.0020            -             -
     └─ Select data returned                                       -  GB            0.0100            -             -
  Total                                                                                               -             -

  aws_s3_bucket.bucket2
  └─ Standard
     ├─ Storage                                                    -  GB            0.0230            -             -
     ├─ PUT, COPY, POST, LIST requests                             -  1k requests   0.0050            -             -
     ├─ GET, SELECT, and all other requests                        -  1k requests   0.0004            -             -
     ├─ Select data scanned                                        -  GB            0.0020            -             -
     └─ Select data returned                                       -  GB            0.0007            -             -
  Total                                                                                               -             -

  aws_s3_bucket_analytics_configuration.bucketanalytics
  └─ Objects monitored                                             -  1M objects    0.1000            -             -
  Total                                                                                               -             -

  aws_s3_bucket_inventory.inventory
  └─ Objects listed                                                -  1M objects    0.0025            -             -
  Total                                                                                               -             -
```

**Pricing details**:

S3:
  * We are adding S3 pricing to the bucket instead of the objects.
  * We are working out which storage classes to show pricing for based on lifecycle rules
  * Object tagging prices are added if any of the lifecycle rules have object tags specified

Route53:
  * Route53 records have no cost if they are aliases to non route53 records.

**Status**:

- [x] Added to resource_registry.go
- [x] Added resource file
- [x] Added integration tests

**Useful links**:

- Pricing:
    - https://aws.amazon.com/route53/pricing/
    - https://aws.amazon.com/s3/pricing/
- Terraform:
    - https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/route53_zone
    - https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/route53_record
    - https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket
    - https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket_inventory
    - https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket_analytics_configuration

### CODE REVIEW

File: internal/providers/terraform/aws/autoscaling_group.go [hunk 1]
```diff
@@ -14,7 +14,8 @@ func GetAutoscalingGroupRegistryItem() *schema.RegistryItem {
 	return &schema.RegistryItem{
 		Name: "aws_autoscaling_group",
 		Notes: []string{
-			"See aws_instance",
+			"S3 replication time control data transfer is not supported by Terraform.",
+			"S3 batch operations is not supported by Terraform.",
```
Comment (alikhajeh1) [inline internal/providers/terraform/aws/autoscaling_group.go:18]:
```suggestion
			"S3 batch operations are not supported by Terraform.",
```

File: internal/providers/terraform/aws/s3_bucket.go [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+		costComponents = append(costComponents, &schema.CostComponent{
+			Name:           "Object tagging",
+			Unit:           "tags",
+			UnitMultiplier: 10000,
+			ProductFilter: &schema.ProductFilter{
+				VendorName: strPtr("aws"),
+				Region:     strPtr(region),
+				Service:    strPtr("AmazonS3"),
+				AttributeFilters: []*schema.AttributeFilter{
+					{Key: "usagetype", ValueRegex: strPtr("/TagStorage-TagHrs/")},
```
Comment (alikhajeh1) [inline internal/providers/terraform/aws/s3_bucket.go:47]: our units is tags but this mentions tagHrs, seems confusing

File: internal/providers/terraform/aws/s3_bucket.go [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+				{Key: "usagetype", ValueRegex: strPtr(fmt.Sprintf("/%s/", usageType))},
+				{Key: "operation", Value: strPtr(operation)},
+			},
+		},
+	}
+}
+
+func s3MonitoringCostComponent(region string) *schema.CostComponent {
+	return &schema.CostComponent{
+		Name:           "Monitoring & automation",
```
Comment (alikhajeh1) [inline internal/providers/terraform/aws/s3_bucket.go:312]:
```suggestion
		Name:           "Monitoring and automation",
```

**Review by alikhajeh1 — APPROVED** [review]

File: internal/providers/terraform/aws/s3_bucket.go (hunk 2 above)
Comment (aliscott) [reply internal/providers/terraform/aws/s3_bucket.go:47]: Yep it's odd - the AWS unit is "Tag-Mo" 🤦

**Pull request merged**